### PR TITLE
additional setup and image build for qos sai test

### PR DESCRIPTION
**Pull request by XuChen-MSFT**
### Description of PR

Summary:
Fixes # (issue)

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [x] 202311

#### What is the motivation for this PR?

add additional setup and image building guide to support qos sai test

#### How did you do it?

add relevant documentation in doc folder

#### How did you verify/test it?

review

### CODE REVIEW

File: docs/testbed/README.testbed.QosRpc.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+```
+
+For instance, to download Broadcom SONiC image for master branch build in pipeline:
+"https://dev.azure.com/mssonic/build/_build/results?buildId=547766&view=results" to replace {buildid} with 547766 and {branchname} with master:
+
+```
+https://sonic-build.azurewebsites.net/ui/sonic/pipelines/1/builds/547766/artifacts?branchName=master
+```
+
+As this is a public service, there is no need to modify [docker_registry.yml](ansible/vars/docker_registry.yml). The QoS SAI test script will automatically pull the syncd rpc container image based on the configuration in [public_docker_registry.yml](ansible/group_vars/all/public_docker_registry.yml):
```
Comment (StormLiangMS) [inline docs/testbed/README.testbed.QosRpc.md:61]: hi @liushilongbuaa  could  you help to comment on this part? Does public build include rpc syncd binary and uploaded as Xu mentioned here?

**Comment by StormLiangMS** [comment]
hi @lipxu  could you take a look?

File: docs/testbed/README.testbed.QosRpc.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+Familiarize yourself with the SONiC build system by reviewing [doc](https://github.com/sonic-net/sonic-buildimage/blob/master/README.md)
+
+## building the syncd RPC Container Image
+
+Follow the build steps in above documentation. Before proceeding to the "build sonic image" step, build the syncd RPC container image for your platforms using below command:
+
+```
+# Build syncd rpc conitainer iamge
+make ENABLE_SYNCD_RPC=y target/docker-syncd-${platform_rpc}-rpc.gz
```
Comment (lipxu) [inline docs/testbed/README.testbed.QosRpc.md:18]: Could we use this CMD to build the rpc container image after proceeding "build sonic image"?

File: docs/testbed/README.testbed.QosRpc.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+- Mellanox: mlnx
+- Nephos: nephos
+
+# Setup your docker registry
+
+For detailed instructions on establishing and managing your Docker registry, refer to [Docker Registry Documentation](https://docs.docker.com/registry/)
+
+# Access docker registry in QoS SAI test script
+
+Modify [docker_registry.yml](ansible/vars/docker_registry.yml) to fill docker registry link, username and password. e.g:
```
Comment (lipxu) [inline docs/testbed/README.testbed.QosRpc.md:35]: (ansible/vars/docker_registry.yml) should be a link in markdown, but seems it is invalid here.

File: docs/testbed/README.testbed.QosRpc.md (hunk 1 above)
Comment (lipxu) [inline docs/testbed/README.testbed.QosRpc.md:61]: (ansible/vars/docker_registry.yml), same with above

Comment (lipxu) [inline docs/testbed/README.testbed.QosRpc.md:61]: (ansible/group_vars/all/public_docker_registry.yml), same with above

File: docs/testbed/README.testbed.QosRpc.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+## understand the build system.
+
+Familiarize yourself with the SONiC build system by reviewing [doc](https://github.com/sonic-net/sonic-buildimage/blob/master/README.md)
+
+## building the syncd RPC Container Image
+
+Follow the build steps in above documentation. Before proceeding to the "build sonic image" step, build the syncd RPC container image for your platforms using below command:
+
+```
+# Build syncd rpc conitainer iamge
```
Comment (Gfrom2016) [inline docs/testbed/README.testbed.QosRpc.md:17]: typo, 'container'

File: docs/testbed/README.testbed.QosRpc.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+```
+docker_registry_host: "soniccr1.azurecr.io"
+
+docker_registry_username: "read-only-user"
+docker_registry_password: "password-of-user"
+```
+
+# Utilize SONiC’s public service for daily syncd RPC container image
+
+If you do not require a private image, recommend to use the SONiC public service to get SONiC images and container image, avoid to build image and setuping docker registry yourself.
```
[the remark this answers is not in the document]
Comment (XuChen-MSFT) [reply docs/testbed/README.testbed.QosRpc.md:47]: updated according to comments

File: docs/testbed/README.testbed.QosRpc.md (hunk 3 above)
Comment (XuChen-MSFT) [reply docs/testbed/README.testbed.QosRpc.md]: updated

File: docs/testbed/README.testbed.QosRpc.md (hunk 4 above)
Comment (XuChen-MSFT) [reply docs/testbed/README.testbed.QosRpc.md]: updated

**Comment by XuChen-MSFT** [comment]
@wangxin  @liushilongbuaa  @xumia any comments?

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Pull request merged**